### ISSUE TITLE
Incorrect Unused Language Extension Warning when using quasiquotes in types

### ISSUE THREAD

**Issue by expipiplus1** [ndmitchell/hlint#237]
```haskell
{-# language QuasiQuotes #-}
type Foo = [q||]
```

HLint produces this incorrect output:
```
Foo.hs:1:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE QuasiQuotes #-}
Why not remove it.
```

**Comment by ndmitchell** [comment]
Thanks for the report - looks like a clear bug. I'll take a look tomorrow morning.

**Comment by ndmitchell** [comment]
I get:

```
Sample.hs:2:12: Error: Parse error: [$q|]
Found:
    {-# language QuasiQuotes #-}
    type Foo = [q||]
  >
```

And looking at https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/template-haskell.html it seems only `[t| ... |]` would be valid in a type splice. Which version of haskell-src-exts are you using? Is that a small test case? If so, do you have the original, which might exhibit the original bug?

**Comment by expipiplus1** [comment]
Ah, I'm pretty sure you're hitting this bug in HSE https://github.com/haskell-suite/haskell-src-exts/issues/289

It's fixed in HEAD, HLint doesn't have any problems compiling against that.

**Comment by ndmitchell** [comment]
Thanks, upgrading to HSE HEAD lets me parse it, and then I fixed the bug. Leaving this ticket open for an HSE upgrade so I can add it as a test.

**Comment by ndmitchell** [comment]
I had to back out the fix since the data type isn't in the released HSE. Once it is released I'll revert a438ba0e45697e65bbcd9fb996d350b07394a45f.

**Comment by expipiplus1** [comment]
Thanks!

**Comment by expipiplus1** [comment]
I think this fix can be reenabled now.

**Comment by ndmitchell** [comment]
Thanks for the reminder. I've reverted it, so should all work now.

**Comment by expipiplus1** [comment]
Thanks
